### PR TITLE
fix: improve DX

### PR DESCRIPTION
**Pull request by ZeroEkkusu**
## Motivation

- Improve the DX
- Fix warnings
- Clean up

## Solution

Various fixes.

### CODE REVIEW

**Review by mds1 — CHANGES_REQUESTED** [review]
Agree with adding this method, just lots of comments around the details and other changes. 

Can we split this into multiple, more tightly scoped PRs? I think one PR for this cheat, and a second for the various cleanup would be a good split. That will make things easier to review, and result in a better diff/auto-generated-changelog for users checking out PRs to see new features

File: test/StdCheats.t.sol
Comment (mds1) [inline test/StdCheats.t.sol:521]: Does `Ct` stand for `Contract`? Let's use a more clear name here also
```suggestion
contract TestContractWithConstructorArgs {
```

File: test/StdCheats.t.sol
Comment (mds1) [inline test/StdCheats.t.sol:525]: We should have another test contract for a non-payable constructor

File: src/StdChains.sol
Comment (ZeroEkkusu) [reply src/StdChains.sol:4]: It's required for unsupported types. However, it must be added in the end-file (`MyTest.sol`) to have effect - adding it to files in the middle (`StdChains.sol`) doesn't do anything when those files don't use the unsupported types.

It should be removed in some other files as well.

File: src/StdCheats.sol
Comment (mds1) [reply src/StdCheats.sol:691]: hmm `etchWithCreationCode` maybe? I think `deployCodeTo` still works though:
- `deployCode` uses CREATE and goes to any address
- `deployCodeTo` uses cheats to place the code at a specific address, so by definition it cannot use CREATE

File: src/StdCheats.sol
Comment (mds1) [reply src/StdCheats.sol:219]: Ah interesting, thanks for the clarification there. So should we just keep the `pure` version around then since that can be used anywhere the `view` version is used?

Comment (ZeroEkkusu) [reply src/StdCheats.sol:219]: `_viewChainId` is used to feed it to `_pureChainId`.

File: src/consoleSafe.sol
Comment (ZeroEkkusu) [reply src/consoleSafe.sol:6]: I'll revert it, but https://github.com/foundry-rs/forge-std/pull/375 is only 3 weeks old. So far, `<Name>Safe` has been the convention.

**Comment by ZeroEkkusu** [comment]
I've removed the `0.6.12` backward compatibility check in the CI because there are no known issues there. @mds1 added it in https://github.com/foundry-rs/forge-std/pull/217/commits/fab939eae5d1c1cf5413b5ac397cd06d07ede9b0; I approved it anyway, not to block the PR.

I'll create a new PR for `deployCodeTo`.

File: test/StdCheats.t.sol
Comment (ZeroEkkusu) [reply test/StdCheats.t.sol]: `constant` must be initialized immediately. I'll add `z` anyway.

File: .github/workflows/ci.yml
Comment (mds1) [inline .github/workflows/ci.yml:24]: I thought we added the 0.6.12 check because the idea was to test the oldest and newest version of each supported minor version, since there are sometimes breaking changes in between. So we had 0.6.2 and 0.6.12 (0.6.2 is when interface inheritance was introduced, which is why we didn't support 0.6.0). Then we had 0.7.0 and 0.7.6 for the same reason. 

But regardless, since these checks are cheap (fast), we may as well just keep it to be safe and lean towards being more conservative?

File: src/StdCheats.sol
Comment (mds1) [inline src/StdCheats.sol:198]: Realizing now this method name has a typo, should be `assumeNotBlacklisted`. To avoid a breaking change we should probably just add that version as an alias. Unrelated to this PR so can handle it separately, just noting here

File: test/StdCheats.t.sol
Comment (mds1) [inline test/StdCheats.t.sol:374]: Then we'd want to change the test name to `assumeNotBlacklisted` also

File: src/StdCheats.sol
Comment (ZeroEkkusu) [reply src/StdCheats.sol:198]: Didn't make the change in this PR.

File: test/StdCheats.t.sol
Comment (ZeroEkkusu) [reply test/StdCheats.t.sol:374]: See above.

File: src/StdCheats.sol
Comment (mds1) [reply src/StdCheats.sol]: Pushed 2feefd2 to make this change

File: test/StdCheats.t.sol
Comment (mds1) [reply test/StdCheats.t.sol]: Pushed 2feefd2 to make this change

**Review by mds1 — APPROVED** [review]
This is awesome, thanks so much @ZeroEkkusu! I added a few small commits (including one to alias `assumeNoBlacklisted` -> `assumeNotBlacklisted`, cc @PaulRBerg) 

CI is passing so I'm going to merge this to unblock #401

**Pull request merged**
